src/cpp/core/include/core/r_util/RTokenCursor.hpp

```diff
@@ -202,14 +209,14 @@ class RTokenCursor
       return currentToken().end();
    }
    
-   const RToken& nextToken() const
+   const RToken& nextToken(std::size_t offset = 1) const
```

Comment:
The `at()` method actually does perform bounds checking, and will return a dummy 'no such token' object if the offset lies outside the bounds.